### PR TITLE
[READY] Blob Mania 1: Reagent Blobs, Split Personality, Storage Blobs, and Blob Broadcast

### PR DESCRIPTION
**Pull request by Iamgoofball**
hrwelec

Blobs now start with a random blob specific reagent.
You can easily add new ones by adding reagents to Blob-Reagents.dm with the path /datum/reagent/blob/reagent_name_here
Blobbernauts now inflict the reagent that the blob starts with instead of "lol infinite burn"
Blobs inflict their reagent on mobs upon expanding onto a tile.
When blobs talk to other blobs over telepathy, it shows their reagent.
Adds "Split consciousness" a 100 cost, node requiring (think blobbernaut) power that produces a second blob overmind (with it's own unique reagent!)
Adds Storage Blobs. Each blob gives you 50 extra max points.
Adds Blob Broadcast.  It's a free verb that will make all your blob spores, zombies, and blobbernauts speak for you. They'll say whatever you want to say, and they'll all say it at once.

This should change up the blob metagame a bit, and allows for more opportunities for blobs to work together on stuff.

Let's say the crew's heavily armored against brute damage and has welders. Have your Acidic Liquid blob buddy sneak in nearby and eliminate their armor while you finish them off with your increased brute damage Skin Ripper reagent.

### CODE REVIEW

**Comment by Razharas** [comment]
Oh wow, now blobs and vines are even closer

**Comment by Iamgoofball** [comment]
:^)

**Comment by Steelpoint** [comment]
So this is a random thing you cannot control?

On the one hand that stops everyone for going to the "meta" choice, on the other hand its pretty random.

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 1]
```diff
@@ -132,14 +132,13 @@
...
+	var/mob/camera/blob/overmind
 
 
 /mob/living/simple_animal/hostile/blobbernaut/AttackingTarget()
 	..()
 	if(isliving(target))
-		var/mob/living/L = target
-		if(L.reagents)
-			L.reagents.add_reagent("spore_burning", 10)
+		overmind.blob_reagent_datum.reaction_mob(target, TOUCH)
```
Comment (RemieRichards) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: you should add a check that the overmind exists.

**Comment by Razharas** [comment]
Stamina blob a shite

File: code/game/gamemodes/blob/overmind.dm [hunk 2]
```diff
@@ -15,18 +15,30 @@
 	var/blob_points = 0
 	var/max_blob_points = 100
 	var/last_attack = 0
+	var/datum/reagent/blob_reagent = "water"
+	var/datum/reagent/blob_reagent_name = "Water"
+	var/datum/reagent/blob_color = ""
```
Comment (RemieRichards) [inline code/game/gamemodes/blob/overmind.dm]: you only need the blob_reagent_datum var, the rest can be obtained using that without separate vars.

File: code/game/gamemodes/blob/powers.dm [hunk 3]
```diff
@@ -265,6 +269,10 @@
 		return
 	last_attack = world.time
 	OB.expand(T, 0)
+	for(var/atom/movable/A in T)
```
Comment (RemieRichards) [inline code/game/gamemodes/blob/powers.dm]: why not just do ```for(var/mob/living/L in T) blob_reagent_datum.reaction_mob(L, TOUCH)```

**Comment by Razharas** [comment]
Things to concider:
* Does brute damage get reduced by armour? If yes then brute blob too weak since other things dont
* Maybe adjust bliblings and blobernaughts to the type of blob?
* Pure stamina damage is just too bad
* Different types dont really change much of anything since apart from brute other damage doesnt get cut by armour
* Those that do act differently namely acid are too OP since they do same damage but also kill items on you

**Comment by Iamgoofball** [comment]
They are greyscaled. It looks better on other colors.
I'll adjust bruteblob.
Blobbernauts are adjusted, it's their nerf.
I'll add some damage stuff to stamina blob.
Different damage types do have effects, toxloss can stop you in important situations to throw up, oxyloss can speed up the crit process and prevent you being saved, and burnloss I think can light you on fire iunno
Acid is supposed to kill items, don't get up close and personal with acid blob. I'll lower it's base brute damage output though.

**Comment by Iamgoofball** [comment]
They are greyscaled. It looks better on other colors.
I'll adjust bruteblob.
Blobbernauts are adjusted, it's their nerf.
I'll add some damage stuff to stamina blob.
Different damage types do have effects, toxloss can stop you in important situations to throw up, oxyloss can speed up the crit process and prevent you being saved, and burnloss I think can light you on fire iunno
Acid is supposed to kill items, don't get up close and personal with acid blob. I'll lower it's base brute damage output though.

**Comment by Iamgoofball** [comment]
whoops

**Comment by Razharas** [comment]
Make core and node icons be black maybe or something
Currently they blend in with usual blob tiles too much

**Comment by Razharas** [comment]
Burn damage cant light you on fire, lighting on fire is special snowflake

**Comment by Razharas** [comment]
By adjusting bloblings and blobbernaughts i mostly meant their damage type

**Comment by RemieRichards** [comment]
@Iamgoofball https://github.com/Iamgoofball/-tg-station/pull/3
I fixed some things I complained about and added the new power discussed in IRC.

**Comment by WJohn** [comment]
While I love the idea, I'm not hugely fond of the 255 red I'm looking at at the picture. I'm hoping you'll tone it down or have some different shades like you did with your cooking PR.

**Comment by paprka** [comment]
The color idea is good and should use the 'alien color' selection like lizards do, but this is a terrible approach to countering riot armor opponents, worse than blobbernauts doing the infinite flame thing.

**Comment by iamgoofballsecondfork** [comment]
dragon drilldongs and testing github bans with cheridan

**Comment by Iamgoofball** [comment]
testing

**Comment by Iamgoofball** [comment]
okay so cloneloss blobs:

they'll come with a buff to cloneloss healing chems

**Comment by Iamgoofball** [comment]
testing

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+		M.apply_damage(10, BRUTE)
+		M.apply_damage(10, BURN)
+		M.adjust_fire_stacks(2)
+		M.IgniteMob()
+		M << "You feel your skin burning and melting!"
+		M.emote("scream")
+
+datum/reagent/blob/lung_destroying_toxin
+	name = "Lung Destroying Toxin"
+	id = "boiling_oil"
```
Comment (Delimusca) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: duplicate ID?

Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: whoops

File: code/game/gamemodes/blob/blobs/core.dm [hunk 5]
```diff
@@ -14,6 +14,9 @@
 	processing_objects.Add(src)
 	if(!overmind)
 		create_overmind(new_overmind)
+		color = overmind.blob_reagent_datum.color
```
Comment (RemieRichards) [inline code/game/gamemodes/blob/blobs/core.dm]: I specifically removed this line as it causes runtimes, the line isn't needed as create_overmind() handles the colour now too. so please revert this back so it's the else chain

**Comment by Atskadan** [comment]
is there a way to greyscale-color a sprite while still retaining actual colors?
if so you could make cores and nodes and stuff still yellow while the rest is whatever color

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+	name = "Boiling Oil"
+	id = "boiling_oil"
+	description = ""
+	color = "#000000"
+
+datum/reagent/blob/boiling_oil/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.adjust_fire_stacks(2)
+		M.IgniteMob()
+		M << "You feel your skin burning!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: It's better descriptions time!
"The blob splashes you with burning oil!"

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+datum/reagent/blob/anti_oxygenation
+	name = "Anti-Oxygenation Liquid"
+	id = "anti_oxygenation"
+	description = ""
+	color = "#00FFFF"
+
+datum/reagent/blob/anti_oxygenation/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.apply_damage(20, OXY)
+		M << "You can't breathe!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: "The blob strikes you, and you feel dizzy and out of breath!"

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+	description = ""
+	color = "#7F0000"
+
+datum/reagent/blob/skin_melter/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.apply_damage(10, BRUTE)
+		M.apply_damage(10, BURN)
+		M.adjust_fire_stacks(2)
+		M.IgniteMob()
+		M << "You feel your skin burning and melting!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: "The blob strikes you, and you feel your skin char and melt!"

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+datum/reagent/blob/explosive // I'm gonna burn in hell for this one
+	name = "Explosive Gelatin"
+	id = "explosive"
+	description = ""
+	color = "#FFA500"
+
+datum/reagent/blob/explosive/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		if(prob(50))
+			M << "The blob explodes!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: "The blob strikes you, and its tendrils explode!"
This should probably happen a touch more often, also.

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+				if((X) && (!istype(X,/mob/living/carbon/human)))
+					X.throw_at(pull)
+				else if(istype(X,/mob/living/carbon/human))
+					var/mob/living/carbon/human/H = X
+					if(istype(H.shoes,/obj/item/clothing/shoes/magboots))
+						var/obj/item/clothing/shoes/magboots/S = H.shoes
+						if(S.magpulse)
+							continue
+					X.throw_at(pull)
+		M << "Everything flies at you!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: "You feel a thrum as the blob strikes you, and everything flies at you!"

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+	name = "Acidic Liquid"
+	id = "blob_acid"
+	description = ""
+	color = "#BD80F4"
+
+datum/reagent/blob/acid/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.acid_act(20,1,10)
+		M.apply_damage(10, BRUTE)
+		M << "You feel your skin and equipment melting off!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: "The blob's tendrils melt though your equipment!"

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+	name = "Lung Destroying Toxin"
+	id = "lung_destroying_toxin"
+	description = ""
+	color = "#00FFC5"
+
+datum/reagent/blob/lung_destroying_toxin/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.apply_damage(10, OXY)
+		M.apply_damage(10, TOX)
+		M << "You feel your lungs rotting!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: The blob strikes you, and your lungs feel heavy and weak!"

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+				end_dir = SOUTH
+			if(M.dir == SOUTH && !end_dir)
+				end_dir = NORTH
+			if(M.dir == EAST && !end_dir)
+				end_dir = WEST
+			if(M.dir == WEST && !end_dir)
+				end_dir = EAST
+			var/atom/throw_target = get_edge_target_turf(M, end_dir)
+			M.throw_at(throw_target, 200, 4)
+			M << "You are sent flying!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: "The blob slams into you, and sends you flying!"

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+		M.apply_effect(10,IRRADIATE,0)
+		if(prob(33))
+			randmuti(M)
+			if(prob(98))
+				randmutb(M)
+			else
+				randmutg(M)
+			domutcheck(M, null)
+			updateappearance(M)
+		M << "You feel irradiated!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: "The blob strikes you, and your skin feels papery and everything hurts!"

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+	name = "Stamina Drainer"
+	id = "stamina_drainer"
+	description = ""
+	color = "#FFFF00"
+
+datum/reagent/blob/stamina_drainer/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.apply_damage(30, STAMINA)
+		M.apply_damage(10, BRUTE)
+		M << "You feel incredibly tired!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: "The blob strikes you, and you feel winded and weak!"

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+datum/reagent/blob/skin_ripper
+	name = "Skin Ripper"
+	id = "skin_ripper"
+	description = ""
+	color = "#FF4C4C"
+
+datum/reagent/blob/skin_ripper/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.apply_damage(20, BRUTE)
+		M << "You feel your skin being ripped off!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: "The blob strikes you, and you feel your skin ripping and tearing off!"

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+datum/reagent/blob/toxic_goop
+	name = "Toxic Goop"
+	id = "toxic_goop"
+	description = ""
+	color = "#008000"
+
+datum/reagent/blob/toxic_goop/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.apply_damage(20, TOX)
+		M << "You feel sick!"
```
Comment (Anonus) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: "The blob strikes you, and you feel sick and nauseated!

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm (hunk 6 above)
Comment (phil235) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: All these messages also needs span classes.

**Comment by Iamgoofball** [comment]
I love it, will change soon

**Comment by AlexanderUlanH** [comment]
Unless I missed some advantage that oxyloss blobs have, oxygen blobs are going to be much less powerful than any of the others.  Not only does their damage heal naturally over time, but there's dexplus, which sets oxyloss to 0 every tick, so if you load up with that, you're completely immune to the blob's attacks.

**Comment by Iamgoofball** [comment]
hm, I'll look into buffing oxyblob

**Comment by Anonus** [comment]
Dex+ is going to make oxyblobs way too weak and very dead even if you give them lexorin, because oxygen damage is stupidly easy to heal away.

**Comment by iamgoofballsecondfork** [comment]
testing testing 123

**Comment by RemieRichards** [comment]
@tgstation/commit-access review pls.

**Comment by TZK13** [comment]
I don't like how difficult it is to see blob structures like nodes and the core, can't you make those into overlays?

**Comment by tkdrg** [comment]
I thought you were going to do something about oxyblob?

**Comment by Iamgoofball** [comment]
my laptop took a huge shit again and it's undergoing repairs

i'll get the sprites modified soon and the overlay shit done and oxyblobs either removed/buffed a shitton

**Comment by Iamgoofball** [comment]
sprites modified

stamina and oxygen blob removed

ready

**Comment by Iamgoofball** [comment]
Enables children blobs to be able to split. Public playtesting can show whether this needs to be disabled.

File: code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+				else if(istype(X,/mob/living/carbon/human))
+					var/mob/living/carbon/human/H = X
+					if(istype(H.shoes,/obj/item/clothing/shoes/magboots))
+						var/obj/item/clothing/shoes/magboots/S = H.shoes
+						if(S.magpulse)
+							continue
+					step_away(X,pull)
+					step_away(X,pull)
+					step_away(X,pull)
+					step_away(X,pull)
```
Comment (Razharas) [inline code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm]: This code piece is so bad i want to cri

**Comment by Razharas** [comment]
3/5 ill give this shit a try

**Comment by RemieRichards** [comment]
@Iamgoofball >makes children blob able to split

oh lordy I prevented this for a reason! Guess it's time for the blob's victims to ghost and join in the /fun/

**Comment by Iamgoofball** [comment]
It can be disabled if it turns into omega blob. I'd like to test run it enabled first.

**Pull request merged**